src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -424,16 +432,19 @@ private TileEntityChest searchRightChestForStack(@NotNull final ItemStack stack)
      * @return the entity of the chest.
      */
     @Nullable
-    private TileEntityChest searchChestWithSimilarItem(final ItemStack stack)
+    private IItemHandler searchItemHandlerWithSimilarItem(final ItemStack stack)
     {
-        for(@NotNull final BlockPos pos : getBuilding().getAdditionalCountainers())
+        for(@NotNull final BlockPos pos : getBuilding().getAdditionalContainers())
         {
             final TileEntity entity = worldObj.getTileEntity(pos);
-            if(entity instanceof TileEntityChest
-                    && InventoryUtils.findFirstSlotInInventoryWith((TileEntityChest) entity, stack.getItem(), -1) != -1
-                    && InventoryUtils.getOpenSlot((TileEntityChest)entity) != -1)
+            if(entity instanceof TileEntityChest)
             {
-                return (TileEntityChest) entity;
+                IItemHandler handler = entity.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'handler' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)